app/classifier/tasks/survey/translations.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import translations from '../../../pages/project/translations';
+
+function SurveyTranslations(props) {
+  const { task } = props;
+  const translation = {
```

Comment:
Maybe try `const translation = Object.assign({}, task)` here, so that translation labels automatically default to the workflow task labels, and the following code can be greatly simplified.

I can't remember if `Object.assign()` does a deep merge of nested objects.

Reply:
Specifically, this needs to be done in such a way that [properties on `translation` aren't pointers to the original properties on `task`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Deep_Clone). 🤔

Reply:
I don't think it does
>The Object.assign() method only copies enumerable and own properties from a source object to a target object.

Reply:
Yeah, the MDN article I linked to says that it creates references, not actual copies. We want to be able to change `translation` without changing `task`.

Reply:
lodash has a deep merge method and we have it as a dependency. I think it might be used in the codebase somewhere already.

Reply:
I'll have a look. I've rewritten the object cloning in cc2114356a8aea888a3a2bbe066b1180c7d44908 to make better use of temp variables and destructuring.

Here, we want to overwrite a key's value if a corresponding key exists in the Panoptes translation resource. The translation wrapper basically gives us something that looks like a task (but contains translated text when available) for backwards compatibility with projects that haven't been migrated over to use translations yet.

I'm also thinking that it might allow is to wrap ZRC components, and pass text into them via a translation prop. Code that doesn't use translations yet should still work by passing a task object as the translation (since the structures are identical) `<Task translation={workflow.tasks[taskKey]} />`.

Reply:
I'm also hoping that these adapters can be ripped out, or greatly simplified, once every Panoptes resource has been migrated across to use translations.

Reply:
Hm, maybe translations is a good case for a generalized higher order component in ZRC?

Reply:
Possibly. I'm finding each resource needs its own, specific adapter component. There isn't much consistency between tasks, for example.

Reply:
If I can figure out a generalised map from `taskStrings` to `translation` then all of this can be turned into a generic adapter for workflow task translations.

Generic translations for all resources might require some reworking of Panoptes internals. I remember @camallen saying that workflow translations have kept the awkward structure that they have here, with the long string keys for tasks, because changing those keys would break anything that relies on them, like classification exports.